### PR TITLE
Add international news article pages

### PR DESCRIPTION
**Pull request by eadpearce**
screencapture-exred-trade-great-8007-international-news-2018-10-18-11_54_14
screencapture-exred-trade-great-8007-international-news-news-article-2018-10-18-11_54_28

### CODE REVIEW

File: conf/urls.py [hunk 1]
```diff
@@ -538,6 +538,16 @@
         prototype.views.NewsArticleDetailView.as_view(),
         name='news-article-detail',
     ),
+    url(
+        r"^international/news/$",
```
Comment (richtier) [inline conf/urls.py]: `international/` or `int/`?

File: prototype/mixins.py [hunk 2]
```diff
@@ -29,3 +31,34 @@ class SlugFromKwargsMixin():
...
     def slug(self):
         return self.kwargs['slug']
+
+
+class InternationalNewsCMSPageMixin(GetCMSPageByFullPathMixin):
+
+    @cached_property
+    def page(self):
+        response = cms_api_client.lookup_by_full_path(
+            full_path=prefix_international_news_url(self.request.path),
```
Comment (richtier) [inline prototype/mixins.py]: you could make this DRYer by adding a property to `GetCMSPageByFullPathMixin` called `cms_lookup_path` that returns `self.request.path`:

```
class GetCMSPageByFullPathMixin(TemplateView):
     @property
     def cms_lookup_path(self):
        return self.request.path

    @cached_property
    def page(self):
        response = cms_api_client.lookup_by_full_path(
            full_path=self.cms_lookup_path,
            draft_token=self.request.GET.get('draft_token'),
        )
        return handle_cms_response(response)
```
then you can do
```
class InternationalNewsCMSPageMixin
    @property
    def cms_lookup_path(self):
        return prefix_international_news_url(self.request.path)
```

that has less duplication of logic :)

File: prototype/mixins.py [hunk 3]
```diff
@@ -29,3 +31,34 @@ class SlugFromKwargsMixin():
...
+            draft_token=self.request.GET.get('draft_token'),
+        )
+        return handle_cms_response(response)
+
+
+class SocialLinksMixin():
+
+    @cached_property
```
Comment (richtier) [inline prototype/mixins.py]: `cached_property` is normally reserved for when the function is performing an "expensive" operation such as retrieving data from an API. It speeds up the code at the tradeoff of adding complexity

this function looks rather cheap, so it's hard to justify the added complexity of using `cached_property`

File: prototype/templates/prototype/sections/services_section.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<section id="services" class="services-section background-stone">
+<section id="services" class="services-section background-stone-30">
```
Comment (richtier) [inline prototype/templates/prototype/sections/services_section.html]: :heart:

File: prototype/templatetags/prototype_tags.py [hunk 5]
```diff
@@ -8,5 +8,5 @@
 @register.filter
 def parse_date(date_string):
     if date_string:
-        return dateparser.parse(date_string).strftime('%m %B %Y')
+        return dateparser.parse(date_string).strftime('%d %B %Y')
```
Comment (richtier) [inline prototype/templatetags/prototype_tags.py]: could you add a test to cover this, please

File: prototype/views.py [hunk 6]
```diff
@@ -53,18 +65,22 @@ def get_context_data(self, *args, **kwargs):
...
+):
+    template_name = 'prototype/domestic_news_detail.html'
     url_name = 'news-article-detail'
+
+
+class InternationalNewsArticleDetailView(
+    NewsArticleDetailView, InternationalNewsCMSPageMixin,
+):
+    template_name = 'prototype/international_news_detail.html'
+    url_name = 'international-news-article-detail'
```
Comment (richtier) [inline prototype/views.py]: how is url_name used?

File: core/sass/partials/_effects.scss [hunk 7]
```diff
@@ -1,189 +0,0 @@
-// EFFECT STYLES
```
Comment (eadpearce) [inline core/sass/partials/_effects.scss]: moved this styling into `_carousel` and `_modal` to make it more clear from the naming what it's for

File: core/sass/partials/_exred-main.scss [hunk 8]
```diff
@@ -1,53 +0,0 @@
-// Start Great Pattern Library
```
Comment (eadpearce) [inline core/sass/partials/_exred-main.scss]: this styling is no longer used

File: prototype/mixins.py (hunk 2 above)
Comment (eadpearce) [reply prototype/mixins.py]: ty i was wondering how i could improve that

File: prototype/mixins.py (hunk 3 above)
Comment (eadpearce) [reply prototype/mixins.py]: okay i wondered whether it was worth it to use that. i can remove it

File: prototype/views.py (hunk 6 above)
Comment (eadpearce) [reply prototype/views.py]: i think it was used in `social_links_builder` but isn't anymore. i can probably remove it

File: conf/urls.py (hunk 1 above)
Comment (eadpearce) [reply conf/urls.py]: international homepage is `international/` so `international/news/` makes sense here. i don't think we use `int/` anywhere

**Pull request merged**